docker-compose.yml

```diff
@@ -1,39 +1,49 @@
 version: "3.3"
 services:
   postgres:
-    image: quay.io/azavea/postgis:2.2-postgres9.5-slim
+    env_file:
+      - .env
     environment:
-      - POSTGRES_DB=district_builder
-      - POSTGRES_USER=district_builder
-      - POSTGRES_PASSWORD=district_builder
+      - POSTGRES_DB=${DATABASE_DATABASE}
```

Comment:
I _think_ this is probably where your data loss is happening -- the environment variable that this is getting interpolated from is [the environment in which `docker-compose` is run](https://docs.docker.com/compose/compose-file/#variable-substitution), which is the VM, and I don't think (based on my reading of the changes here) that the VM has this available, so I think you're setting these all to empty strings.

Reply:
That might cause other issues also. There's one sort of straightforward fix, which would be sourcing `.env` as the last step of provisioning, since you tell people to copy `.env.sample` before they run `./scripts/setup` anyway

Reply:
The variable are making their way into the `docker-compose.yml` okay. All that you need to do is have a `.env` file in the same directory: https://docs.docker.com/compose/environment-variables/#the-env-file

I verified it with `docker-compose config`:
```
vagrant@vagrant-ubuntu-trusty-64:/vagrant$ docker-compose config
services:
  celery:
    build:
      context: /vagrant/django/publicmapping
    command:
    - worker
    - --app=publicmapping
    - --uid=reporter
    - --loglevel=INFO
    entrypoint: /usr/local/bin/celery
    environment:
      ADMIN_EMAIL: admin@admin.com
      ADMIN_PASSWORD: admin
      ADMIN_USER: admin
      DATABASE_DATABASE: district_builder
      DATABASE_HOST: postgres.internal.districtbuilder.com
      DATABASE_PASSWORD: district_builder
      DATABASE_PORT: '5432'
      DATABASE_USER: district_builder
      DEBUG: "True"
      KEY_VALUE_STORE_DB: '0'
      KEY_VALUE_STORE_HOST: redis.internal.districtbuilder.com
      KEY_VALUE_STORE_PASSWORD: redis_password
      KEY_VALUE_STORE_PORT: '6379'
      MAP_SERVER_ADMIN_PASSWORD: admin
      MAP_SERVER_ADMIN_USER: admin
      MAP_SERVER_HOST: geoserver.internal.districtbuilder.com
      MAP_SERVER_PASSWORD: geoserver
      MAP_SERVER_PORT: '9091'
      MAP_SERVER_USER: geoserver
      WEB_APP_PASSWORD: django
      WEB_APP_PORT: '8080'
    links:
    - postgres:postgres.internal.districtbuilder.com
    - redis:redis.internal.districtbuilder.com
    volumes:
    - /vagrant/django/publicmapping:/usr/src/app:rw
    - reports:/opt/reports:rw
  config:
    build:
      context: /vagrant/django/publicmapping
    command:
    - generate_settings.py
    - config/config.xsd
    - config/config.xml
    entrypoint: /usr/local/bin/python
    volumes:
    - /vagrant/django/publicmapping:/usr/src/app:rw
  django:
    build:
      context: /vagrant/django/publicmapping
    command:
    - --workers=2
    - --timeout=60
    - --bind=0.0.0.0:8080
    - --reload
    - --log-level=debug
    - --access-logfile=-
    - --error-logfile=-
    - -kgevent
    - publicmapping.wsgi
    depends_on:
    - config
    entrypoint: /usr/local/bin/gunicorn
    environment:
      ADMIN_EMAIL: admin@admin.com
      ADMIN_PASSWORD: admin
      ADMIN_USER: admin
      DATABASE_DATABASE: district_builder
      DATABASE_HOST: postgres.internal.districtbuilder.com
      DATABASE_PASSWORD: district_builder
      DATABASE_PORT: '5432'
      DATABASE_USER: district_builder
      DEBUG: "True"
      KEY_VALUE_STORE_DB: '0'
      KEY_VALUE_STORE_HOST: redis.internal.districtbuilder.com
      KEY_VALUE_STORE_PASSWORD: redis_password
      KEY_VALUE_STORE_PORT: '6379'
      MAP_SERVER_ADMIN_PASSWORD: admin
      MAP_SERVER_ADMIN_USER: admin
      MAP_SERVER_HOST: geoserver.internal.districtbuilder.com
      MAP_SERVER_PASSWORD: geoserver
      MAP_SERVER_PORT: '9091'
      MAP_SERVER_USER: geoserver
      WEB_APP_PASSWORD: django
      WEB_APP_PORT: '8080'
    expose:
    - '8080'
    links:
    - postgres:postgres.internal.districtbuilder.com
    - redis:redis.internal.districtbuilder.com
    - geoserver:geoserver.internal.districtbuilder.com
    volumes:
    - /vagrant/django/publicmapping:/usr/src/app:rw
    - /vagrant/sld:/opt/sld:rw
    - data:/data:rw
    - reports:/opt/reports:rw
  geoserver:
    build:
      context: /vagrant/geoserver
    environment:
      ADMIN_EMAIL: admin@admin.com
      ADMIN_PASSWORD: admin
      ADMIN_USER: admin
      DATABASE_DATABASE: district_builder
      DATABASE_HOST: postgres.internal.districtbuilder.com
      DATABASE_PASSWORD: district_builder
      DATABASE_PORT: '5432'
      DATABASE_USER: district_builder
      DEBUG: "True"
      KEY_VALUE_STORE_DB: '0'
      KEY_VALUE_STORE_HOST: redis.internal.districtbuilder.com
      KEY_VALUE_STORE_PASSWORD: redis_password
      KEY_VALUE_STORE_PORT: '6379'
      MAP_SERVER_ADMIN_PASSWORD: admin
      MAP_SERVER_ADMIN_USER: admin
      MAP_SERVER_HOST: geoserver.internal.districtbuilder.com
      MAP_SERVER_PASSWORD: geoserver
      MAP_SERVER_PORT: '9091'
      MAP_SERVER_USER: geoserver
      WEB_APP_PASSWORD: django
      WEB_APP_PORT: '8080'
    links:
    - postgres:postgres.internal.districtbuilder.com
    ports:
    - published: 9091
      target: 8080
    volumes:
    - /vagrant/geoserver/data_dir:/data:rw
    - /vagrant/sld:/data/sld:rw
  nginx:
    command: /bin/bash -c "envsubst < /etc/nginx/conf.d/default.template > /etc/nginx/conf.d/default.conf
      && nginx -g 'daemon off;'"
    depends_on:
    - django
    environment:
      ADMIN_EMAIL: admin@admin.com
      ADMIN_PASSWORD: admin
      ADMIN_USER: admin
      DATABASE_DATABASE: district_builder
      DATABASE_HOST: postgres.internal.districtbuilder.com
      DATABASE_PASSWORD: district_builder
      DATABASE_PORT: '5432'
      DATABASE_USER: district_builder
      DEBUG: "True"
      KEY_VALUE_STORE_DB: '0'
      KEY_VALUE_STORE_HOST: redis.internal.districtbuilder.com
      KEY_VALUE_STORE_PASSWORD: redis_password
      KEY_VALUE_STORE_PORT: '6379'
      MAP_SERVER_ADMIN_PASSWORD: admin
      MAP_SERVER_ADMIN_USER: admin
      MAP_SERVER_HOST: geoserver.internal.districtbuilder.com
      MAP_SERVER_PASSWORD: geoserver
      MAP_SERVER_PORT: '9091'
      MAP_SERVER_USER: geoserver
      WEB_APP_PASSWORD: django
      WEB_APP_PORT: '8080'
    image: nginx:1.12
    ports:
    - published: 8080
      target: 8080
    volumes:
    - /vagrant/nginx:/etc/nginx/conf.d:rw
    - /vagrant/django/publicmapping/static:/opt/static:rw
    - /vagrant/sld:/opt/sld:rw
    - reports:/opt/reports:rw
  postgres:
    environment:
      ADMIN_EMAIL: admin@admin.com
      ADMIN_PASSWORD: admin
      ADMIN_USER: admin
      DATABASE_DATABASE: district_builder
      DATABASE_HOST: postgres.internal.districtbuilder.com
      DATABASE_PASSWORD: district_builder
      DATABASE_PORT: '5432'
      DATABASE_USER: district_builder
      DEBUG: "True"
      KEY_VALUE_STORE_DB: '0'
      KEY_VALUE_STORE_HOST: redis.internal.districtbuilder.com
      KEY_VALUE_STORE_PASSWORD: redis_password
      KEY_VALUE_STORE_PORT: '6379'
      MAP_SERVER_ADMIN_PASSWORD: admin
      MAP_SERVER_ADMIN_USER: admin
      MAP_SERVER_HOST: geoserver.internal.districtbuilder.com
      MAP_SERVER_PASSWORD: geoserver
      MAP_SERVER_PORT: '9091'
      MAP_SERVER_USER: geoserver
      POSTGRES_DB: district_builder
      POSTGRES_PASSWORD: district_builder
      POSTGRES_USER: district_builder
      WEB_APP_PASSWORD: django
      WEB_APP_PORT: '8080'
    healthcheck:
      interval: 15s
      retries: 3
      test:
      - CMD-SHELL
      - PGPASSWORD=district_builder
      - pg_isready
      - -U
      - district_builder
      timeout: 5s
    image: quay.io/azavea/postgis:2.2-postgres9.5-slim
  redis:
    command: redis-server --requirepass redis_password
    environment:
      ADMIN_EMAIL: admin@admin.com
      ADMIN_PASSWORD: admin
      ADMIN_USER: admin
      DATABASE_DATABASE: district_builder
      DATABASE_HOST: postgres.internal.districtbuilder.com
      DATABASE_PASSWORD: district_builder
      DATABASE_PORT: '5432'
      DATABASE_USER: district_builder
      DEBUG: "True"
      KEY_VALUE_STORE_DB: '0'
      KEY_VALUE_STORE_HOST: redis.internal.districtbuilder.com
      KEY_VALUE_STORE_PASSWORD: redis_password
      KEY_VALUE_STORE_PORT: '6379'
      MAP_SERVER_ADMIN_PASSWORD: admin
      MAP_SERVER_ADMIN_USER: admin
      MAP_SERVER_HOST: geoserver.internal.districtbuilder.com
      MAP_SERVER_PASSWORD: geoserver
      MAP_SERVER_PORT: '9091'
      MAP_SERVER_USER: geoserver
      WEB_APP_PASSWORD: django
      WEB_APP_PORT: '8080'
    image: redis:3.2-alpine
    ports:
    - published: 6379
      target: 6379
version: '3.3'
volumes:
  data: {}
  reports: {}
```
Plus if everything was blank nothing would work because hosts and ports are defined there.

Reply:
Interesting. Nevermind then!

Reply:
But, relatedly, I didn't push this commit to properly map the environment variables until recently (after @ddohler started testing): https://github.com/PublicMapping/DistrictBuilder/pull/502/commits/9944ba8518cf5209ca35beb55248d0f0bb103bad

My theory is that before pushing that commit, we were connecting to the wrong database, whatever the default was.